src/wallet/scriptpubkeyman.cpp

```diff
@@ -1185,6 +1186,7 @@ bool LegacyScriptPubKeyMan::ReserveKeyFromKeyPool(int64_t& nIndex, CKeyPool& key
             throw std::runtime_error(std::string(__func__) + ": keypool entry invalid");
         }
 
+        m_reserved_key_to_index[nIndex] = keypool.vchPubKey.GetID();
```

Comment:
In commit "Key pool: Change ReturnDestination interface to take address instead of key" (06d411488c1c669f00a1c947416267941dbb7f7b)

This commit needs a better description. The current handwavy placeholder text doesn't explain very much. Suggest replacing it with the explanation in your PR description "In order for ScriptPubKeyMan to be generic..."

Reply:
Updated the commit message.